cpp/src/cuda_stream.cpp

```diff
@@ -22,10 +22,11 @@
 
 namespace rmm {
 
-cuda_stream::cuda_stream()
-  : stream_{[]() {
+cuda_stream::cuda_stream(cuda_stream::flags flags)
+  : stream_{[flags]() {
               auto* stream = new cudaStream_t;  // NOLINT(cppcoreguidelines-owning-memory)
-              RMM_CUDA_TRY(cudaStreamCreate(stream));
+              // TODO: use std::to_underlying once C++23 is allowed.
+              RMM_CUDA_TRY(cudaStreamCreateWithFlags(stream, static_cast<unsigned int>(flags)));
```

Comment:
I like the TODO, let's use the formally-equivalent (though more longwinded) thing here.
```suggestion
              RMM_CUDA_TRY(cudaStreamCreateWithFlags(stream, static_cast<std::underlying_type_t<cuda_stream::flags>>(flags)));
```

Reply:
Thanks, done.